File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        
        int write = 0;
        int temp;

        for(int read = 0; read<nums.length; read++){
            if(nums[read] != 0){
                temp = nums[write];
                nums[write] = nums[read];
                nums[read] = temp;
                write++;
            }
        }

    }
}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        idx = 0

        for i in range(len(nums)):
            if nums[i] != 0:
                temp = nums[idx]
                nums[idx] = nums[i]
                nums[i] = temp
                idx+=1
        
        return nums
        